roles/rds-apb-openshift/tasks/main.yml

```diff
@@ -24,10 +24,12 @@
       Application: ansible-rds
       Description: "{{ ansible_user_id }} RDS {{ db_engine }} {{ tds }}"
   register: rds
+  when: state=="present"
 
 - name: retrieve rds endpoint IP
   shell: "getent hosts {{ rds.instance.endpoint }} | awk '{ print $1 }'"
   register: rds_endpoint_ip
+  when: state=="present"
 
 - name: create openshift service
   k8s_v1_service:
```

Comment:
You can simply add `state: {{ state }}` here and it will be made absent on deprovision.